cmd/geth/retesteth.go

```diff
@@ -0,0 +1,880 @@
+		if err := api.mineBlock(); err != nil {
+			return false, err
+		}
+	}
+	fmt.Printf("Mined %d blocks\n", number)
+	return true, nil
+}
+
+func (api *RetestethAPI) mineBlock() error {
+	parentHash := rawdb.ReadCanonicalHash(api.ethDb, api.blockNumber)
+	parent := rawdb.ReadBlock(api.ethDb, parentHash, api.blockNumber)
+	var timestamp uint64
+	if api.blockInterval == 0 {
+		timestamp = uint64(time.Now().Unix())
+	} else {
+		timestamp = parent.Time() + api.blockInterval
+	}
+	header := &types.Header{
+		ParentHash: parent.Hash(),
+		Number:     big.NewInt(int64(api.blockNumber + 1)),
+		GasLimit:   core.CalcGasLimit(parent, 0, 8000000),
+		Extra:      api.extraData,
+		Time:       timestamp,
+	}
+	header.Coinbase = api.author
+	if api.engine != nil {
+		api.engine.Prepare(api.blockchain, header)
+	}
+	// If we are care about TheDAO hard-fork check whether to override the extra-data or not
+	if daoBlock := api.chainConfig.DAOForkBlock; daoBlock != nil {
+		// Check whether the block is among the fork extra-override range
+		limit := new(big.Int).Add(daoBlock, params.DAOForkExtraRange)
+		if header.Number.Cmp(daoBlock) >= 0 && header.Number.Cmp(limit) < 0 {
+			// Depending whether we support or oppose the fork, override differently
+			if api.chainConfig.DAOForkSupport {
+				header.Extra = common.CopyBytes(params.DAOForkBlockExtra)
+			} else if bytes.Equal(header.Extra, params.DAOForkBlockExtra) {
+				header.Extra = []byte{} // If miner opposes, don't let it use the reserved extra-data
+			}
+		}
+	}
+	statedb, err := api.blockchain.StateAt(parent.Root())
+	if err != nil {
+		return err
+	}
+	if api.chainConfig.DAOForkSupport && api.chainConfig.DAOForkBlock != nil && api.chainConfig.DAOForkBlock.Cmp(header.Number) == 0 {
+		misc.ApplyDAOHardFork(statedb)
+	}
+	gasPool := new(core.GasPool).AddGas(header.GasLimit)
+	txCount := 0
+	var txs []*types.Transaction
+	var receipts []*types.Receipt
+	var coalescedLogs []*types.Log
+	if gasPool.Gas() >= params.TxGas {
+	outer:
+		for address := range api.txSenders {
+			m := api.txMap[address]
+			for nonce := statedb.GetNonce(address); ; nonce++ {
+				if tx, ok := m[nonce]; ok {
+					// Try to apply transactions to the state
+					statedb.Prepare(tx.Hash(), common.Hash{}, txCount)
+					snap := statedb.Snapshot()
+
+					receipt, _, err := core.ApplyTransaction(
+						api.chainConfig,
+						api.blockchain,
+						&api.author,
+						gasPool,
+						statedb,
+						header, tx, &header.GasUsed, *api.blockchain.GetVMConfig(),
+					)
+					if err != nil {
+						statedb.RevertToSnapshot(snap)
+						continue outer
+					}
+					txs = append(txs, tx)
+					receipts = append(receipts, receipt)
+					coalescedLogs = append(coalescedLogs, receipt.Logs...)
+					delete(m, nonce)
+					if len(m) == 0 {
+						// Last tx for the sender
+						delete(api.txMap, address)
+						delete(api.txSenders, address)
+					}
+					txCount++
+					if gasPool.Gas() < params.TxGas {
+						break outer
```

Comment:
I would personally rather have a check in the outer loop, something like `if blockFull { break } `, instead of using a label. We don't usually use labels

Reply:
Thank you, fixed